runtime/src/message_processor.rs

```diff
@@ -58,10 +59,36 @@ fn get_subset_unchecked_mut<'a, T>(
         .collect())
 }
 
+pub struct PreAccount {
+    pub debitable: bool,
+    pub lamports: u64,
+    pub data_len: usize,
+    pub data: Option<Vec<u8>>,
+    pub owner: Pubkey,
+    pub executable: bool,
+    pub rent_epoch: Epoch,
+}
+impl PreAccount {
+    pub fn new(a: &Account, debitable: bool, program_id: &Pubkey) -> Self {
+        Self {
+            debitable,
+            lamports: a.lamports,
+            data_len: a.data.len(),
+            // Don't copy data if not needed
+            data: if *program_id != a.owner || !debitable {
```

Comment:
`!debitable` is hard to read, can this function run on `readonly`, or `creditonly`?

Reply:
This is extended from `message.is_debitable()`,   Review the naming starting from the root., how about in a different PR?  In this PR I can put it back as `is_debitable` is that helps clarify.